### PR TITLE
improved feedback description handling

### LINKED ISSUE #134: [Bug] Submitting feedback via the feedbackform strips newlines
**Describe the bug**
When feedback is submitted via the feedbackform, 

**To Reproduce**
1. Go to the feedback button
2. enter 
image

3. See #132 

**Expected behavior**
The `\n` should be replaced with `\n  ` in the backend, to be more in line what I expect a user to expect

**Additional context**
I opened this as an issue, to get some feedback if my view on this is valid, or if we should stick with the current status quo

### PR DESCRIPTION
**Pull request by CommanderStorm**
added a improved feedback description parsing and feedback description tests

resolves #134

### CODE REVIEW

File: server/src/feedback.rs [hunk 1]
```diff
@@ -213,10 +214,50 @@ fn parse_request(req_data: &Json<FeedbackPostData>) -> (String, String, Vec<Stri
...
+        assert_eq!(clean_feedback_data("a\nb  \nc", 9), "a  \nb  \nc");
+        assert_eq!(clean_feedback_data("a      \nb", 9), "a  \nb");
+        assert_eq!(clean_feedback_data("a\n\nb", 9), "a  \n  \nb");
+        assert_eq!(clean_feedback_data("a\n   b", 9), "a  \n   b");
+    }
+    #[test]
+    fn truncate_len() {
+        for i in 0..10 {
+            assert_eq!(clean_feedback_data("abcd", 0), "");
+        }
```
Comment (octycs) [inline server/src/feedback.rs:252]: Is there a reason this is in a `for`?

Comment (CommanderStorm) [reply server/src/feedback.rs:252]: Ups.. reason was that I wanted to reduce my code-footprint :)

What I really meant is:
```suggestion
        for i in 0..10 {
            assert_eq!(clean_feedback_data("abcd", i), "abcd".truncate(i));
        }
```

File: server/src/feedback.rs [hunk 2]
```diff
@@ -213,10 +214,50 @@ fn parse_request(req_data: &Json<FeedbackPostData>) -> (String, String, Vec<Stri
...
+        assert_eq!(clean_feedback_data("a\n   b", 9), "a  \n   b");
+    }
+    #[test]
+    fn truncate_len() {
+        for i in 0..10 {
+            assert_eq!(clean_feedback_data("abcd", i), "abcd".truncate(i));
+        }
+        assert_eq!(clean_feedback_data("abcd", 1), "a");
+        assert_eq!(clean_feedback_data("abcd", 3), "abc");
+        assert_eq!(clean_feedback_data("abcd", 6), "abcd");
```
Comment (octycs) [inline server/src/feedback.rs:255]: Should we leave these then?

Comment (CommanderStorm) [reply server/src/feedback.rs:255]:
```suggestion
```

**Pull request merged**